PyStemmusScope/bmi/variable_reference.py

```diff
@@ -151,4 +160,31 @@ class BmiVariable:
         grid=0,
         keys=["GroundwaterSettings", "topLevel"],
     ),
+    BmiVariable(
```

Comment:
I add the `GroundwaterSettings.gw_dep` variable for the cases when we have groundwater observations (no MODFLOW). In that case, the user will use `GroundwaterSettings.gw_dep` instead of `GroundwaterSettings.topLevel` and `GroundwaterSettings.headBotmLayer`. 

I am not sure if `GroundwaterSettings.gw_dep` can be treated as input without changes in the code or not. Normally, `GroundwaterSettings.gw_dep` is calculated [here](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/1c15e9a7c5bef374469c4d0f4f04dcb9397d47d0/src/STEMMUS_SCOPE.m#L354C1-L354C113). Do we need to do changes in the code to let the `GroundwaterSettings.gw_dep` be treated as input instead of calculated?